corehq/apps/user_importer/validation.py

```diff
@@ -513,30 +547,18 @@ def _validate_uploading_user_access(self, spec):
 
         # 2. Ensure the user is only adding the user to/removing from *new locations* that they have permission
         # to access.
-        if 'location_code' in spec:
-            locs_being_assigned = self._get_locs_being_assigned(spec)
+        if locs_ids_being_assigned:
```

Comment:
Thank you for catching that bug, fixed it  [a5ee207](https://github.com/dimagi/commcare-hq/pull/35448/commits/a5ee2070e26869c807406c7c234104833dad4b76)

I think this is equivalent - I do the `location_code' in spec` within `validate_spec` and only do the validations if that's true. For the snippet referenced here, I removed the conditional check so that validation will always run.

Reply:
Oh dang, thanks for catching that. [8fee5fe](https://github.com/dimagi/commcare-hq/pull/35448/commits/8fee5fe9462dd5d79b35c6ac5837d01cb0797eff) should fix it. Although I added some nasty looking variable and function names that i'm open to renaming.